lib/puppet/type/file/source.rb

```diff
@@ -276,7 +276,11 @@ def chunk_file_from_disk
     end
 
     def get_from_puppet_source(source_uri, &block)
-      request = Puppet::Indirector::Request.new(:file_content, :find, source_uri, nil, :environment => resource.catalog.environment_instance)
+      if content_uri = resource[:content_uri]
+        request = Puppet::Indirector::Request.new(:static_file_content, :find, content_uri, nil, :code_id => resource.catalog.code_id)
```

Comment:
Just to close the loop on what happened in Hipchat: Ticket lied to me about it being needed. It is apparently required, and I've got an updated version for this